src/App/Services/Http.js

```diff
@@ -1,22 +1,21 @@
-import {Observable} from "rxjs"
+import {from, of, throwError} from "rxjs";
+import {map, mergeMap} from "rxjs/operators"
+let reactiveFetch  = (url, props) => from(fetch(url,{
+    ...props
+}))
 
+let middlware = mergeMap((response)=>
+     response.status == 200 || response.status ==304 ? of(response) : throwError(response)
+)
 
-class Http {
-    get (url) {
-        return Observable.fromPromise(fetch(url));
-    }
-
-    getBody(url) {
-        return Observable.fromPromise(fetch(url).then((e) => e.json()))
-    }
-
-    post(url, props) {
-        return Observable.fromPromise(fetch(url, {
-                method: "POST",
-                ...props
-            }
-        ))
-    }
-}
-
-module.exports = new Http();
\ No newline at end of file
+module.exports = (()=>({
```

Comment:
there is no need for an IIFE?
also, you could have used the export keyword to export default {...} or explicitly export each function